data/types/base.go

```diff
@@ -68,6 +69,26 @@ type Base struct {
 	VersionInternal   int                           `json:"-" bson:"_version,omitempty"`
 }
 
+// custom DeviceTime formatting
+type DeviceTime time.Time
+
+func (t DeviceTime) MarshalJSON() ([]byte, error) {
```

Comment:
You can implement the [TextMarshaller](https://pkg.go.dev/encoding#TextMarshaler) interface here:
```
type DeviceTime time.Time

func (d DeviceTime) MarshalText() (text []byte, err error) {
	return []byte(time.Time(d).Format("2006-January-02")), nil
}
```